src/signed_data.rs

```diff
@@ -861,28 +922,28 @@ mod tests {
         general_purpose::STANDARD.decode(&base64).unwrap()
     }
 
-    static SUPPORTED_ALGORITHMS_IN_TESTS: &[&signed_data::SignatureAlgorithm] = &[
+    static SUPPORTED_ALGORITHMS_IN_TESTS: &[&dyn signed_data::SignatureAlgorithm] = &[
         // Reasonable algorithms.
-        &signed_data::ECDSA_P256_SHA256,
-        &signed_data::ECDSA_P384_SHA384,
-        &signed_data::ED25519,
+        signed_data::ECDSA_P256_SHA256,
+        signed_data::ECDSA_P384_SHA384,
+        signed_data::ED25519,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PKCS1_2048_8192_SHA256,
+        signed_data::RSA_PKCS1_2048_8192_SHA256,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PKCS1_2048_8192_SHA384,
+        signed_data::RSA_PKCS1_2048_8192_SHA384,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PKCS1_2048_8192_SHA512,
+        signed_data::RSA_PKCS1_2048_8192_SHA512,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PKCS1_3072_8192_SHA384,
+        signed_data::RSA_PKCS1_3072_8192_SHA384,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+        signed_data::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PSS_2048_8192_SHA384_LEGACY_KEY,
+        signed_data::RSA_PSS_2048_8192_SHA384_LEGACY_KEY,
         #[cfg(feature = "alloc")]
-        &signed_data::RSA_PSS_2048_8192_SHA512_LEGACY_KEY,
+        signed_data::RSA_PSS_2048_8192_SHA512_LEGACY_KEY,
         // Algorithms deprecated because they are annoying (P-521) or because
         // they are nonsensical combinations.
-        &signed_data::ECDSA_P256_SHA384, // Truncates digest.
-        &signed_data::ECDSA_P384_SHA256, // Digest is unnecessarily short.
+        signed_data::ECDSA_P256_SHA384, // Truncates digest.
```

Comment:
The comment above mentions P-521, but that's not actually defined here -- maybe update the comment while you're here?